### ISSUE TITLE
Too-close villages not consistently valid/invalid

### ISSUE THREAD

**Issue by GrifterMage** [TangoTek/TekTopia-Community#177]
If a new town hall marker is placed too close to an existing, loaded village, it won't be valid and will not glow. However, if it's kept up, it's possible for that "new" village to become valid (under some unknown circumstances), which then invalidates the "old" village.

**Comment by TangoTek** [comment]
Its first come first serve.  The first town hall loaded will be the active one.  If the active on is unloaded, they'll switch.

The only chance I can think of making is to put a time stamp on each town hall and always let the oldest one win, but... that seems like a pretty odd thing to do.

Suggestions?

**Comment by JohnyDL** [comment]
Maybe do it not on oldest but on largest number of tokens brought, that way the most established town with the most active play time is the winner, seems sucky for the loser either way but I think it's probably better for the smallest least established village to relocate even if they were there first because there's less to move and less to lose. There will be cases where a casual player is ousted by an active one, but so long as everyone has done a reasonable check for near by villages and isn't being malicious and using this technique to grief which is banable for other reason I think that it's the fairest way.

**Comment by GrifterMage** [comment]
Instead of searching through loaded entities for town halls, could you create an external list of villages to query? The file would be updated whenever a new town hall is validated/broken, similar to villages.dat worked for vanilla villages. That way it won't matter which chunks happen to be loaded.

**Comment by JohnyDL** [comment]
I'm not sure how you'd persist that unless you put something in the spawn chunks to maintain that list Grifter, but I do think that there could be a more robust way to do the 400x400 town border perhaps a banner in every chunk on the border? When a new village is made that conflicts with the 400x400 boarder the banner line follows the halfway points between the villages and where 3 or 4 meet well the furthest is ignored... okay explained with a picture:
![Villages](https://user-images.githubusercontent.com/5202568/58746099-330a2b80-8452-11e9-84e0-0009857332f3.png)
1px = 1 block, grey squares are chunks the big black squares are the +-200 village boundary, but the saturated coloured lines are where the actual borders are. there're construction lines to show roughly how I made the image. 
The towns could each get their own random banner too so you know where your space ends, and since these banners are on the outer borders a few things happen. First players can see the towns as they approach and know not to build inside there near is fine (a new village will distort the boundaries around it, but inside is a "don't bother village won't work here". Second, you don't need to load the town halls to know where they, are each banner could keep their town hall's origin point as NBT data and scanning out for these banners should take a lot less time than scanning out for the single town marker since you don't have to look in every chunk at every entity you and could probably do it by sampling 100-150 or so chunks (the boarders and a handful near the centre) rather than all the chunks. Third I think the maths is cool and would make me happy XD

**Comment by Jamcdonald120** [comment]
@johnydl you can make anything persist chunk loading in code, you just increase the scope of the variable

**Comment by JohnyDL** [comment]
~~Yeah but chunk loading adds server overhead and would mean that every town hall chunk would be a partially loaded village at all times, my suggestion was to reduce the amount of overhead 99.9% of the time by placing/removing banners in the world the 0.1% of the time a town hall token is added/removed from an item frame. Keeping towns partially loaded means that villagers might starve you have to do processing on those chunks when the player isn't there.~~

~~I would like to see a "Chunk load villages" option for single player or small multiplayer worlds but it's probably better to use a chunkloading Mod than to add it to Tango's~~

"Make anything persist chunk loading" was miss interpreted as "make anything able to keep chunks loaded" apparently this isn't what was said.

**Comment by Jamcdonald120** [comment]
No, thats not how programing works, you can keep an oct tree of town halls and check for near by villages without ever asking minecraft core to do anything, let alone loading chunks.  The worst way to do it would be to force load and then scan every entity (item frame) within 400 blocks each time a town hall marker is placed

**Comment by JohnyDL** [comment]
Tango asked for the chat to stop so I'm not going to argue with you about this.
I offered an alternative that I thought enhanced the mod and might enhance player experiences.
You and me arguing about "how programming works" from theory and no actual code is pointless
It's up to Tango to decide if it fits in with how he's programmed it so far.
